packages/core/src/types.ts

```diff
@@ -1349,3 +1350,25 @@ export interface Behavior<TEvent extends EventObject, TEmitted = any> {
   initialState: TEmitted;
   start?: (actorCtx: ActorContext<TEvent, TEmitted>) => TEmitted;
 }
+
+export type EventsFrom<T> = EventObject &
+  (T extends StateMachine<any, any, infer TEvents0, any>
+    ? TEvents0
+    : T extends Model<any, infer TEvents1, any>
+    ? TEvents1
+    : T extends State<any, infer TEvents2, any, any>
+    ? TEvents2
+    : T extends Interpreter<any, any, infer TEvents3, any>
+    ? TEvents3
+    : never);
+
+export type ContextFrom<T> = object &
```

Comment:
what `object &` brings to the table here? 🤔

Reply:
Restricts the resulting type to an object; is there a better way?

Reply:
Since all the slots you are inferring should be restricted to something (like an `object`), you shouldn't worry about it here - an incompatible type should never be created in the first place (it wouldn't compile). And if an incompatible type can be created then your restriction here is not correct anyway, as in - it could create problems down the road.

Reply:
1568e0ac9a84135c06be2ed54f01d5005dd70d7d